create_kubernetes_sysext.sh

```diff
@@ -0,0 +1,99 @@
+curl -o kubelet -fsSL "https://dl.k8s.io/v${VERSION}/bin/linux/${ARCH}/kubelet"
+
+rm -rf "${SYSEXTNAME}"
+mkdir -p "${SYSEXTNAME}"/usr/bin
+mv kubectl kubeadm kubelet "${SYSEXTNAME}"/usr/bin
+
+chmod +x "${SYSEXTNAME}"/usr/bin/{kubectl,kubeadm,kubelet}
+
+# setup kubelet service.
+mkdir -p "${SYSEXTNAME}/usr/lib/systemd/system"
+cat > "${SYSEXTNAME}/usr/lib/systemd/system/kubelet.service" <<-'EOF'
+[Unit]
+Description=kubelet: The Kubernetes Node Agent
+Documentation=https://kubernetes.io/docs/home/
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+ExecStart=/usr/bin/kubelet
+Restart=always
+StartLimitInterval=0
+RestartSec=10
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+mkdir -p "${SYSEXTNAME}/usr/lib/systemd/system/kubelet.service.d"
+cat > "${SYSEXTNAME}/usr/lib/systemd/system/kubelet.service.d/10-kubeadm.conf" <<-'EOF'
+# Note: This dropin only works with kubeadm and kubelet v1.11+
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
+Environment="KUBELET_CONFIG_ARGS=--config=/var/lib/kubelet/config.yaml"
+# This is a file that "kubeadm init" and "kubeadm join" generates at runtime, populating the KUBELET_KUBEADM_ARGS variable dynamically
+EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
+# This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
+# the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
+EnvironmentFile=-/etc/sysconfig/kubelet
+ExecStart=
+ExecStartPre=/usr/bin/mkdir -p /opt/libexec /opt/libexec.work
+ExecStartPre=/usr/bin/cp -r /usr/local/bin/cni/ /opt/bin/cni
+ExecStartPre=/usr/bin/cp /usr/local/share/kubernetes-version /etc/kubernetes-version
+ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_CONFIG_ARGS $KUBELET_KUBEADM_ARGS $KUBELET_EXTRA_ARGS
+EOF
+
+mkdir -p "${SYSEXTNAME}/usr/local/share/"
+echo "v${VERSION}" > "${SYSEXTNAME}/usr/local/share/kubernetes-version"
+
+cat > "${SYSEXTNAME}/usr/lib/systemd/system/usr-libexec.mount" <<-'EOF'
+[Unit]
+Description=Kubernetes flex volume plugin directory
+
+[Mount]
+What=overlay
+Where=/usr/libexec
+Type=overlay
+Options=lowerdir=/usr/libexec,workdir=/opt/libexec.work,upperdir=/opt/libexec
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+mkdir -p "${SYSEXTNAME}/usr/lib/systemd/system/multi-user.target.d"
+{ echo "[Unit]"; echo "Upholds=kubelet.service"; } > "${SYSEXTNAME}/usr/lib/systemd/system/multi-user.target.d/10-kubelet-service.conf"
+{ echo "[Unit]"; echo "Upholds=usr-libexec.mount"; } > "${SYSEXTNAME}/usr/lib/systemd/system/multi-user.target.d/10-usr-libexec.conf"
+
+# install CNI.
+# TODO: use CNI version / arch variable
+curl -o cni.tgz -fsSL https://github.com/containernetworking/plugins/releases/download/v1.3.0/cni-plugins-linux-amd64-v1.3.0.tgz
```

Comment:
```suggestion
curl -o cni.tgz -fsSL https://github.com/containernetworking/plugins/releases/download/v1.3.0/cni-plugins-linux-${ARCH}-v1.3.0.tgz
```
We should also require a second version for this unless you can query the version from some other manifest.

Reply:
It also works to download the latest version but I think an explicit parameter is better.

Reply:
FYI the github URL for `latest` has it before the `download` part:
https://github.com/containernetworking/plugins/releases/latest/download/cni-plugins-linux-amd64-v1.3.0.tgz